dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java

```diff
@@ -75,14 +77,26 @@ public abstract class SourceCodeRepoInterface {
      * First gets all the file names, then see if any of them matches the README regex
      * @param repositoryId
      * @param branch
+     * @param overrideLocation if present, use this location instead of the root
      * @return
      */
-    public String getREADMEContent(String repositoryId, String branch) {
-        List<String> strings = this.listFiles(repositoryId, "/", branch);
-        if (strings == null) {
-            return null;
+    public String getREADMEContent(String repositoryId, String branch, String overrideLocation) {
+
+        Optional<String> first;
+        if (!Strings.isNullOrEmpty(overrideLocation)) {
+            final Path overridePath = Paths.get(overrideLocation);
+            List<String> strings = this.listFiles(repositoryId, overridePath.getParent().toString(), branch);
```

Comment:
Since `overrideLocation` is a file path, we might be able to simplify by replacing this block (lines 87-92) with:
```
   first = Optional.of(overrideLocation);
```

Reply:
If I understand correctly, that may work, but I think its less efficient from a cache perspective. 

i.e. if the user specifies `/foo/readme.md` the existing code will list the files in `/foo` (hopefully cached or has other files in it) and then look for `readme.md` inside it. 
The simplification would directly look for `/foo/readme.md` regardless of whether there is anything in `/foo`. 

(Arguably, a further improvement would be to list the files in `/` as well to see if `/foo` exists ...)

Reply:
Understood regarding the caching.  If we leave the semantics as is, maybe we could make this section easier to read by moving the file search logic into an `existsFile` method?